src/aggregated_bls/party_i.rs

```diff
@@ -52,11 +53,45 @@ impl Keys {
 
     pub fn combine_local_signatures(sigs: &[SIG]) -> BLSSignature {
         let (head, tail) = sigs.split_at(1);
-        let sig_sum = tail.iter().fold(head[0], |x, acc| x + acc);
+        let sig_sum = tail.iter().fold(head[0], |acc, x| acc + x);
         BLSSignature { sigma: sig_sum }
     }
 
     pub fn verify(signature: &BLSSignature, message: &[u8], apk: &APK) -> bool {
         signature.verify(message, apk)
     }
+
+    pub fn batch_aggregate_bls(sig_vec: &[BLSSignature]) -> BLSSignature {
+        let (head, tail) = sig_vec.split_at(1);
+        BLSSignature {
+            sigma: tail.iter().fold(head[0].sigma, |acc, x| acc + x.sigma),
+        }
+    }
+
+    fn core_aggregate_verify(apk_vec: &[APK], msg_vec: &[&[u8]], sig: &BLSSignature) -> bool {
```

Comment:
I am not currently doing this, and am unsure if it is needed to check that `sig` is an element of the subgroup, and for each `apk` if it is a valid public key as in: 
https://tools.ietf.org/html/draft-irtf-cfrg-bls-signature-04#section-2.9.